### ISSUE TITLE
Using UIM with Anki creates defunct processes

### ISSUE THREAD

**Issue by qsniyg** [uim/uim#102]
Using the development version of UIM with Anki creates defunct processes each time I switch to the Anki window (one defunct process when I switch to it, another when I switch away).

If I run anki with `QT_IM_MODULE= anki`, there are no defunct processes.

**Comment by qsniyg** [comment]
I should add that I use a text entry function for many of my cards (see https://www.youtube.com/watch?v=5tYObQ3ocrw ).

However, even for a card that has no text entry, simply switching to the next card can lead to 46 extra defunct processes.

**Comment by qsniyg** [comment]
Anki just got updated under arch linux (which uses PyQt5 instead of, previously, PyQt4), and UIM no longer works, period. I have the latest git version of UIM installed, with Qt5 enabled, and every other Qt5 (and PyQt5) program I have used has worked just fine.

After rolling back to the previous version of Anki (which uses PyQt4 with Python 2), UIM works again, although with the aforementioned defunct process issue.

Interestingly enough, both native and PyQt4 applications (other than Anki) work just fine, without any defunct processes. 

I'll file a bug report to Anki as well, because it seems to be something specific about Anki that is causing the problem.

**Comment by qsniyg** [comment]
This is a problem with Qt4/5's WebEngine (and is not specific to Anki). While most Qt4/5 applications work fine, those that use QtWebEngine encounter the issues I described above.

I will file a bug report to Qt on this issue as well.

**Comment by foghawk** [comment]
@qsniyg

I've been having persistent issues with uim and Qt5 on Arch Linux: sometime between mid-October of 2018 and mid-April of this year, uim stopped working in all Qt5 applications (not just those depending on QtWebEngine). Based on the timing I suspect Qt 5.12, but Qt [supposedly fixed](https://bugreports.qt.io/browse/QTBUG-62962) this in 5.9 and I haven't been able to find any more recent reports.

(According to the comments on that issue, the problem was actually that Arch's PKGBUILD didn't have `--with-qt5 --with-qt5-immodule`, but [that's been fixed](https://git.archlinux.org/svntogit/packages.git/commit/?h=packages/uim&id=29ab91824da508c0455e3dacafdbbbd566a985b3).)

Is that your bug report I found? Do you have uim working in Qt5/Arch now?

**Comment by qsniyg** [comment]
@foghawk Yeah that was my bug report :) I haven't actually used any QtWebEngine application for a long time, so I'm not sure if this bug still exists or not. Do you still experience issues with UIM?

**Comment by foghawk** [comment]
@qsniyg Yes—but in _all_ Qt5 applications, not just QtWebEngine-based. Can you verify whether uim is working for you? (The uim package includes `uim-pref-qt5`, which has some text fields to check with, if you don't have any other Qt5 applications installed.)

**Comment by qsniyg** [comment]
@foghawk I haven't had any issues with Qt5 applications for quite a long time... I'm not sure how I fixed it though, it's been years :(
